:sparkles: introducing file-permissions hook

dderjoel opened the pull request:
## Created Hook
- [x] Hook Name *(if applicable)*: file permissions
- [x] Added tests
- [x] Used gitmoji for commits

**Description of Hook:**
>  This hooks allows the user to specify desired file permissions to be set, after the `tsc` command finished, e.g. to make the main js file executable.
 
**Description of Edits**
> Description here&hellip;

swimauger commented:
Hey @dderjoel this looks great! I just published a dev version with these changes and everything seems to work as intended! My only comment is if we could change the config option from `tsc-hooks-permissions` to camel case/maybe just rename it to `permissions` or `filePermissions`. So long as we don't add anything that screws up the tsconfig [schema](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/tsconfig.json), we should be okay to rename it to this.

dderjoel commented:
Hi, I 
[x] changed the property to `filePermissions`
[x] added documentation in the table in the Readme
[x] added information, that hooks are executed in-order
[x] changed the HTML to MD-layout (feel free to omit this commit, if you prefer HTML
[x] added an example for the file-permissions hook

dderjoel commented:
Let me know, if there is something missing for a merge to tsc-hooks/main

swimauger commented:
Hey @dderjoel can you merge the latest version of the main branch into your fork?
